tcpchecker.go

```diff
@@ -0,0 +1,129 @@
+package checkup
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"time"
+)
+
+// TCPChecker implements a Checker for TCP endpoints.
+type TCPChecker struct {
+	// Name is the name of the endpoint.
+	Name string `json:"endpoint_name"`
+
+	// URL is the URL of the endpoint.
+	URL string `json:"endpoint_url"`
+
+	// TLSEnabled controls whether to enable TLS or not.
+	// If set, TLS is enabled.
+	TLSEnabled bool `json:"tls,omitempty"`
+
+	// TLSVerify controls whether to validate server
+	// TLS certificate or not.
+	TLSVerify bool `json:"tls_verify,omitempty"`
```

Comment:
Maybe we should invert this so that the default is to verify? Maybe `TLSSkipVerify` or something.